libdeno/deno.h

```diff
@@ -25,15 +25,26 @@ typedef struct deno_s Deno;
 typedef void (*deno_recv_cb)(void* user_data, int32_t req_id,
                              deno_buf control_buf, deno_buf data_buf);
 
+// A callback to implement ES Module imports. User must call deno_resolve_ok()
+// at most once during deno_resolve_cb. If deno_resolve_ok() is not called, the
+// specifier is considered invalid and will issue an error in JS. The reason
+// deno_resolve_cb does not return deno_module is to avoid unnecessary heap
```

Comment:
>  does not return deno_module is to avoid unnecessary heap allocations

Did it seem very expensive? I doubt this is even in the vicinity of a hot path.

Reply:
Not so much about performance, but rather about keeping memory management simple. I don't want to have to think about allocating something in Rust and then passing ownership to libdeno